### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by spikerheado1234**
Sets up integration testing with reference to ticket number #357.

### CODE REVIEW

**Comment by lewisdaly** [comment]
@spikerheado1234 can you please fix the test artifacts not uploading? Take a look at the `Uploading Artifacts` step here: https://app.circleci.com/pipelines/github/mojaloop/auth-service/399/workflows/9de1900f-63b9-4ffd-b4eb-027e76b62cc0/jobs/2933/steps

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,24 @@
...
+# Create a non-root user: as-user
+RUN adduser -D as-user 
+USER as-user
+COPY --chown=as-user --from=builder /opt/auth-service .
 # cleanup
-RUN apk del build-dependencies
-# RUN npm prune --production
-
-RUN npm run test:unit
+#RUN apk del build-dependencies
```
Comment (lewisdaly) [inline Dockerfile:21]: Please remove commented out code. You will also likely want to run the `npm prune`

File: Dockerfile [hunk 2]
```diff
@@ -1,32 +1,24 @@
...
+#RUN npm prune --production
 EXPOSE 4004
-CMD ["npm", "run", "start"]
+CMD ["npm", "run", "start"]
```
Comment (lewisdaly) [inline Dockerfile:24]: Hmm maybe we need an editorconfig file in this repo or something... there probably should be a newline here

File: docker-compose.yml [hunk 3]
```diff
@@ -8,6 +8,7 @@ services:
   auth-service:
     container_name: as_auth-service
     image: mojaloop/auth-service:local
+    command: sh -c "/opt/auth-service/scripts/wait4.js my-sql"
```
Comment (lewisdaly) [inline docker-compose.yml:11]: Don't you want to run the server after completing this command? Take a look at `pisp/docker-local/docker-compose.yml`

File: src/shared/config.ts [hunk 4]
```diff
@@ -39,6 +39,7 @@ export interface ServiceConfig {
   PARTICIPANT_ID: string;
 
   // ../lib/config.
+  ENVIRONMENT: string;
```
Comment (lewisdaly) [inline src/shared/config.ts:42]: Hmm I don't like generic, superpowered config name like `ENVIRONMENT`. Is there a better one you can think of? Or if this is just to configure the database connection, we should maybe rely on users mounting their own knex file like we do in other projects.

**Comment by spikerheado1234** [comment]
> @spikerheado1234 can you please fix the test artifacts not uploading? Take a look at the `Uploading Artifacts` step here: https://app.circleci.com/pipelines/github/mojaloop/auth-service/399/workflows/9de1900f-63b9-4ffd-b4eb-027e76b62cc0/jobs/2933/steps

Done @lewisdaly!

**Pull request merged**